controllers/eventhandlers/serviceimport.go

```diff
@@ -54,53 +56,60 @@ func (h *enqueueRequestsForServiceImportEvent) Generic(e event.GenericEvent, que
 }
 
 func (h *enqueueRequestsForServiceImportEvent) enqueueImpactedService(queue workqueue.RateLimitingInterface, serviceImport *mcs_api.ServiceImport) {
-	glog.V(6).Infof("enqueueImpactedHTTPRoute, serviceImport[%v]\n", serviceImport)
+	h.log.Infof("enqueueImpactedRoute, serviceImport[%v]", serviceImport)
 
-	httpRouteList := &gateway_api.HTTPRouteList{}
+	var routes []core.Route
 
-	h.client.List(context.TODO(), httpRouteList)
+	switch h.routeType {
+	case core.HttpRouteType:
+		routes = core.ListHTTPRoutes(h.client, context.TODO())
+	case core.GrpcRouteType:
+		routes = core.ListGRPCRoutes(h.client, context.TODO())
+	default:
+		h.log.Errorf("Invalid routeType %s", h.routeType)
+	}
 
-	for _, httpRoute := range httpRouteList.Items {
-		if !isServiceImportUsedByHTTPRoute(httpRoute, serviceImport) {
+	for _, route := range routes {
+		if !isServiceImportUsedByRoute(route, serviceImport) {
 			continue
 		}
 
-		glog.V(6).Infof("enqueueRequestsForServiceImportEvent --> httproute %v\n", httpRoute)
+		h.log.Infof("enqueueRequestsForServiceImportEvent: route name %s, namespace %s",
+			route.Name(), route.Namespace())
+
 		namespacedName := types.NamespacedName{
-			Namespace: httpRoute.Namespace,
-			Name:      httpRoute.Name,
+			Namespace: route.Namespace(),
+			Name:      route.Name(),
 		}
 
 		queue.Add(reconcile.Request{
 			NamespacedName: namespacedName,
 		})
-
 	}
-
 }
 
-func isServiceImportUsedByHTTPRoute(httpRoute gateway_api.HTTPRoute, serviceImport *mcs_api.ServiceImport) bool {
-	for _, httpRule := range httpRoute.Spec.Rules {
-		for _, httpBackendRef := range httpRule.BackendRefs {
-			if string(*httpBackendRef.BackendObjectReference.Kind) != "serviceimport" {
+func isServiceImportUsedByRoute(route core.Route, serviceImport *mcs_api.ServiceImport) bool {
+	for _, rule := range route.Spec().Rules() {
+		for _, backendRef := range rule.BackendRefs() {
+			if string(*backendRef.Kind()) != "serviceimport" {
```

Comment:
Should be
`if string(*backendRef.Kind()) != "Serviceimport" {`

Someone mentioned this bug, could you fix in passing?

https://github.com/aws/aws-application-networking-k8s/pull/326/files

Reply:
Will do, thanks for pointing it out. Found it in target group model builder too.